File: src/main/java/nl/devpieter/lobstar/models/whitelist/WhitelistEntry.java
package nl.devpieter.lobstar.models.whitelist;

import org.jetbrains.annotations.Nullable;

import java.util.UUID;

public record WhitelistEntry(
        UUID id,
        UUID issuerId,

        UUID playerId,
        @Nullable UUID serverId,

        boolean isWhitelisted
) {
}

File: src/main/java/nl/devpieter/lobstar/models/whitelist/WhitelistEntry.kt
package nl.devpieter.lobstar.models.whitelist;

import java.util.*

data class WhitelistEntry(
    val id: UUID,

    val issuerId: UUID,
    val playerId: UUID,
    val serverId: UUID?,

    var isWhitelisted: Boolean,
    var isSuperEntry: Boolean,

    var hasExpiration: Boolean,
    var expirationDate: Date?
)